group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -24387,6 +24387,268 @@ public void testShareGroupRecordsNoExceptionOnReplay(CoordinatorRecord record) {
         assertDoesNotThrow(() -> context.replay(record));
     }
 
+    @Test
+    public void testConsumerGroupAssignmentResolvesWithCompaction() {
+        String groupId = "fooup";
+        String memberA = "memberA";
+        String memberB = "memberB";
+
+        Uuid topicId = Uuid.randomUuid();
+        String topicName = "foo";
+
+        CoordinatorMetadataImage metadataImage = new MetadataImageBuilder()
+            .addTopic(topicId, topicName, 2)
+            .addRacks()
+            .buildCoordinatorMetadataImage();
+        long topicHash = computeTopicHash(topicName, metadataImage);
+
+        MockPartitionAssignor assignor = new MockPartitionAssignor("range");
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .withConfig(GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_CONFIG, List.of(assignor))
+            .withConsumerGroup(new ConsumerGroupBuilder(groupId, 10)
+                .withMetadataHash(topicHash))
+            .build();
```

Comment:
We can remove most of this setup and let the `replay` method create the group. The group lookup will have to be moved to after the `replay`s.
```suggestion
        CoordinatorMetadataImage metadataImage = new MetadataImageBuilder()
            .addTopic(topicId, topicName, 2)
            .buildCoordinatorMetadataImage();

        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
            .withMetadataImage(metadataImage)
            .build();
```